scripts/ci/test_ids/test_id_checking_lib.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2025 3mdeb <contact@3mdeb.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import difflib
+import re
+import subprocess
+
+from robot.model import SuiteVisitor
+from robot.running import TestSuiteBuilder
+
+# Maybe use os-config/environment-test-ids.py instead???
+# A mapping from id to some name is missing though
+# and the bootmanager entries might not be perfect to
+# insert into the braces in test names
+OS_IDS = {
+    r"\.201": r"\(Ubuntu\)",
+    r"\.301": r"\(Windows\)",
+    r"\.202": r"\(Fedora\)",
+}
+
+
+class TestCasesFinder(SuiteVisitor):
+    def __init__(self):
+        self.tests = []
+
+    def visit_test(self, test):
+        self.tests.append(test)
+
+
+def get_test_cases_from_dir(directory):
+    builder = TestSuiteBuilder()
+    try:
+        testsuite = builder.build(directory)
+    except Exception as e:
+        print(
+            f"Error building test suite from {directory}: {e}. Assuming no test cases."
+        )
+        return []
+    finder = TestCasesFinder()
+    testsuite.visit(finder)
+
+    list_of_tests = finder.tests
+
+    return list_of_tests
+
+
+def get_id(test):
+    res = re.search("^([A-Z]{0,9}[0-9]{3}\\.[0-9]{3}).*", test.name)
```

Comment:
I don't think we want to have shorter identifiers.

```suggestion
    res = re.search("^([A-Z]{3,9}[0-9]{3}\\.[0-9]{3}).*", test.name)
```